src/sagemaker/workflow/pipeline.py

```diff
@@ -310,7 +307,12 @@ def start(
             update_args(kwargs, PipelineParameters=parameters)
             return self.sagemaker_session.sagemaker_client.start_pipeline_execution(**kwargs)
         update_args(kwargs, PipelineParameters=format_start_parameters(parameters))
-        response = self.sagemaker_session.sagemaker_client.start_pipeline_execution(**kwargs)
+
+        # retry on AccessDeniedException to cover case of tag propagation delay
+        response = retry_with_backoff(
+            lambda: self.sagemaker_session.sagemaker_client.start_pipeline_execution(**kwargs),
+            botocore_client_error_code="AccessDeniedException",
+        )
```

Comment:
The default retry attempt of this method is 8
So the max wait time is: 0 + 1 + ... 2^7 = 2^8 - 1 = 255 secs ~= 4 mins, which should be acceptable